src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -190,7 +212,7 @@ BOOST_AUTO_TEST_CASE(DBSCANEpsilonNoisePointsTest)
 
   for (size_t i = 0; i < output1.n_elem; ++i)
   {
-    if (output1[i] == 18446744073709551615)
+    if (arma::accu(output1[i] > inputData.n_cols))
     {
       noisePointsOutput1++;
```

Comment:
I think this will always set noisePointOutput to a max of ,1 as you are just checking the return value of accu in if statement. I think if we want number of noisePoints then something like `noisePoints = arma::accu(output1[i] > inputData.n_cols) `should be enough. And you may impose the condition for negative garbage values too.